transient-aws/hive-example/dispatch.sh

```diff
@@ -0,0 +1,316 @@
+#!/bin/bash
+#
+# (c) Copyright 2015 Cloudera, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+RANDOM_TOKEN=$(cat /dev/urandom | env LC_CTYPE=C tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
```

Comment:
Catting /dev/urandom seems undesirable to me. If you just want a random token, you could do any of:

```
uuidgen | tr -d '-'
$(( $RANDOM * 32768 + $RANDOM )) # just digits
dd bs=18 count=1 if=/dev/urandom 2>/dev/null | base64 | tr +/ _. # includes underscore and period
```